planner/core/rule_inject_extra_projection.go

```diff
@@ -40,18 +41,23 @@ func (pe *projInjector) inject(plan PhysicalPlan) PhysicalPlan {
 	for _, child := range plan.Children() {
 		pe.inject(child)
 	}
+
 	switch p := plan.(type) {
 	case *PhysicalHashAgg:
```

Comment:
This two case can also be combined.

Reply:
It seems that we can not merge these two types into one `case` branch though they hold the same attribute.

Reply:
Oh, yeah. If we want to do that, we need something like `basePhysicalAgg`.